octavia_proxy/api/drivers/elbv3/driver.py

```diff
@@ -559,19 +589,31 @@ def flavors(self, session, project_id, query_filter=None):
         query_filter.pop('project_id', None)
 
         result = []
-
-        for fl in session.vlb.flavors(**query_filter):
-            fl_data = _flavors.FlavorResponse.from_sdk_object(fl)
-            fl_data.provider = PROVIDER
-            result.append(fl_data)
-
+        if 'name' in query_filter:
+            query_filter['name'] = ''.join(
+                ('L7_flavor.elb.', query_filter['name'])
+            )
+        if 'id' in query_filter:
+            fl_data = self.flavor_get(
+                project_id=project_id, session=session,
+                fl_id=query_filter['id']
+            )
+            if fl_data:
+                result.append(fl_data)
+        else:
+            for fl in session.vlb.flavors(**query_filter):
+                if not fl['name'].startswith('L4'):
```

Comment:
let's be friendly to anybody reading the code and use `if fl['name'].startswith('L4_flavor.elb'):` cause that will correspond to the replacement in the create_lb